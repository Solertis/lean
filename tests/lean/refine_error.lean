constant p {α : Type} [has_zero α] : α → Prop
constant pax {α : Type} [has_zero α] : p (0:α)

lemma ex : p (0:nat) :=
@pax int _
