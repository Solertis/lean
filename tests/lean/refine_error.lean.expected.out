refine_error.lean:5:1: error: type mismatch, expression
  @pax ℤ int.has_zero
has type
  @p ℤ int.has_zero 0
but is expected to have type
  @p ℕ nat.has_zero 0
